src/simcore_service_deployment_agent/auto_deploy_task.py

```diff
@@ -354,17 +380,18 @@ def setup(app: web.Application):
 
 
 async def background_task(app: web.Application):
-    app[TASK_STATE] = State.STARTING
+    app["state"] = {TASK_NAME: State.STARTING}
     app[TASK_NAME] = asyncio.get_event_loop().create_task(auto_deploy(app))
     yield
     task = app[TASK_NAME]
     task.cancel()
+    await task
```

Comment:
this will trigger a ```asyncio.CancelledError``` exception. Is it catched later on? if not it should be in a try...except or contextlib.suppress